src/components/layout/Quiz/index.js

```diff
@@ -0,0 +1,77 @@
+import React from "react"
+import { Redirect } from "react-router-dom"
+import { connect } from "react-redux"
+import PropTypes from "prop-types"
+
+import Compass from "../../../images/compass.png"
+
+// Redux
+import { setLocation } from "../../../actions/quizActions"
+
+// Material UI
+import {
+  Box,
+  Button,
+  CardMedia,
+  Container,
+  Typography,
+} from "@material-ui/core"
+
+// Components
+import StateQuestion from "./StateQuestion"
+
+const Quiz = ({ quiz, setLocation }) => {
+  if (!quiz.agreement) {
+    return <Redirect to="/" />
+  }
+
+  return (
+    <Container maxWidth="lg">
+      <Box display={"flex"} alignItems={"center"} mb={2}>
+        <CardMedia src={Compass}>
+          <img src={Compass} />
+        </CardMedia>
+
+        <Box ml={2}>
+          <Typography variant={"h1"}>Compensation Compass</Typography>
+        </Box>
+      </Box>
+      {quiz.question == 0 ? <StateQuestion /> : null}
```

Comment:
The idea here is that the `null` would be replaced with a component that generates a quiz question from information from the backend.

Reply:
OK cool! If we're going to do that, we might as well go all the way and just render a `<Question>` component here that takes in `quiz.question` as a prop and then renders `<StateQuestion />` or `<CustomQuestion />`.

I would love more context on how the backend decides which quiz question is next. Does this just have to do with what state they pick, so that we can serve questions specific to the state? Or is there more complex logic going on, like presenting questions in a different order depending on answers? 

I'm not sure if it makes sense given what the logic is, but it could be nice to use a constant like `quiz.question == LOCATION_QUESTION` so that we don't have to depend on an index. If the questions always come in the same order, but may just differ in content depending on the state, indices could still be OK, but I could see it being clearer to know in the code exactly what question we're talking about.

What do you think?

Reply:
I think it might be better to stick with how @egiurleo has it, just to keep the scope in line with just performing the first quiz question. Also, since we are utilizing redux, we won't necessarily be passing props between components as most of the components can be connected and won't need to be dumb.

For the order of quiz questions, it will actually be randomized, but each quiz question will be mapped to a quiz object prop by id so that if 1) is now 4), they still map within the redux store to the same "vote" by the id, if that makes sense. At least, that's how I've been envisioning it. We'll likely randomize the questions within the redux store as another action once state is picked and the "Next" button clicked as just an array of ids - that way we know the order through that specific quiz set and can use it as a sort of navigation.

My understanding is all questions are the same for all states, but not all resources are the same - as some states may have different resources eligible.

Reply:
I'm going to leave it as is for now, but that's a good call out, Alex, and I'll definitely poke whoever ends up working on the rest of the questions to keep this in mind.

The reason I chose indices is because, as Grant said, I was imagining pulling the quiz questions down from the backend and then storing them in some random order in the redux store. There are definitely other ways to do it that may be better for this use case, so I'm not attached to my implementation :)